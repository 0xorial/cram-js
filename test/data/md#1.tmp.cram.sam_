@HD	VN:1.0	SO:coordinate
@SQ	SN:a	LN:40	M5:e970f0ddfe4128e554ec1550bfe17678	UR:/Users/rbuels/dev/htslib/test/md.fa
x	0	a	1	255	40M	*	0	0	AAAAAAAAAACCCCCCCCYNNRGGGGGGGGTTTTTTTTTT	*	NM:i:2	MD:Z:19N0N19
x*	0	a	1	255	40M	*	0	0	CAAAAAAAANNCCCCCCCYNNRGGGGGGGGTTTTTTTTTC	*	NM:i:6	MD:Z:0A8A0C8N0N18T0
xIP	0	a	1	255	1I10M1I10M2I10M2P3I2P10M1I	*	0	0	NAAAAAAAAAAGCCCCCCCCCCAAGGGGGGGGGGCCCTTTTTTTTTTN	*	MD:Z:18Y0N0N0R18	NM:i:12
xIP*	0	a	1	255	1I10M1I10M2I10M2P3I2P10M1I	*	0	0	NCAAAAAAAAAGCCCCCCCCCCAAGGGGGGGGGACCCATTTTTTTTCN	*	MD:Z:0A17Y0N0N0R7G0T8T0	NM:i:16
xD	0	a	1	255	9M1D8M2D7M3D6M	*	0	0	AAAAAAAAACCCCCCYNNRGGGGGTTTTTT	*	NM:i:9	MD:Z:9^A6C0C0^YN0N6^GGG6
xD*	0	a	1	255	9M1D8M2D7M3D6M	*	0	0	CAAAAAAANNCCCCCCTTGGGGGGTTTTTA	*	MD:Z:0A7A0^A0C6C0^YN0N0R5^GGG5T0	NM:i:13
xN	0	a	1	255	10M20N10M	*	0	0	AAAAAAAAAATTTTTTTTTT	*	MD:Z:20	NM:i:0
xN*	0	a	1	255	10M20N10M	*	0	0	CAAAAAAACCAATTTTTTTA	*	MD:Z:0A7A0A0T0T7T0	NM:i:6
xS	0	a	11	255	5H10S20M10S5H	*	0	0	AAAAAAAAAACCCCCCCCYNNRGGGGGGGGTTTTTTTTTT	*	NM:i:2	MD:Z:9N0N9
xS*	0	a	11	255	5H10S20M10S5H	*	0	0	CAAAATAAAACCCCCCCCYNNRGGGGGGGGTTTTATTTTC	*	NM:i:2	MD:Z:9N0N9
