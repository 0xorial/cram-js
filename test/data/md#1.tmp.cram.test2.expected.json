[
  {
    "tags": {
      "NM": 2,
      "MD": "19N0N19"
    },
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 40,
    "alignmentStart": 1,
    "readGroupId": -1,
    "readName": "x",
    "templateSize": 0,
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 0
  },
  {
    "tags": {
      "NM": 6,
      "MD": "0A8A0C8N0N18T0"
    },
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 40,
    "alignmentStart": 1,
    "readGroupId": -1,
    "readName": "x*",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "X",
        "position": 1,
        "data": 0
      },
      {
        "operator": "X",
        "position": 10,
        "data": 3
      },
      {
        "operator": "X",
        "position": 11,
        "data": 3
      },
      {
        "operator": "X",
        "position": 40,
        "data": 1
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 1
  },
  {
    "tags": {},
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 48,
    "alignmentStart": 1,
    "readGroupId": -1,
    "readName": "xIP",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "i",
        "position": 1,
        "data": "N"
      },
      {
        "operator": "i",
        "position": 12,
        "data": "G"
      },
      {
        "operator": "X",
        "position": 21,
        "data": 1
      },
      {
        "operator": "X",
        "position": 22,
        "data": 1
      },
      {
        "operator": "I",
        "position": 23,
        "data": "AA"
      },
      {
        "operator": "X",
        "position": 25,
        "data": 2
      },
      {
        "operator": "X",
        "position": 26,
        "data": 2
      },
      {
        "operator": "P",
        "position": 35,
        "data": 2
      },
      {
        "operator": "I",
        "position": 35,
        "data": "CCC"
      },
      {
        "operator": "P",
        "position": 38,
        "data": 2
      },
      {
        "operator": "i",
        "position": 48,
        "data": "N"
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 2
  },
  {
    "tags": {},
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 48,
    "alignmentStart": 1,
    "readGroupId": -1,
    "readName": "xIP*",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "i",
        "position": 1,
        "data": "N"
      },
      {
        "operator": "X",
        "position": 2,
        "data": 0
      },
      {
        "operator": "i",
        "position": 12,
        "data": "G"
      },
      {
        "operator": "X",
        "position": 21,
        "data": 1
      },
      {
        "operator": "X",
        "position": 22,
        "data": 1
      },
      {
        "operator": "I",
        "position": 23,
        "data": "AA"
      },
      {
        "operator": "X",
        "position": 25,
        "data": 2
      },
      {
        "operator": "X",
        "position": 26,
        "data": 2
      },
      {
        "operator": "X",
        "position": 34,
        "data": 0
      },
      {
        "operator": "P",
        "position": 35,
        "data": 2
      },
      {
        "operator": "I",
        "position": 35,
        "data": "CCC"
      },
      {
        "operator": "P",
        "position": 38,
        "data": 2
      },
      {
        "operator": "X",
        "position": 38,
        "data": 0
      },
      {
        "operator": "X",
        "position": 47,
        "data": 1
      },
      {
        "operator": "i",
        "position": 48,
        "data": "N"
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 3
  },
  {
    "tags": {
      "NM": 9,
      "MD": "9^A6C0C0^YN0N6^GGG6"
    },
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 30,
    "alignmentStart": 1,
    "readGroupId": -1,
    "readName": "xD",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "D",
        "position": 10,
        "data": 1
      },
      {
        "operator": "B",
        "position": 16,
        "data": [
          "Y",
          255
        ]
      },
      {
        "operator": "X",
        "position": 17,
        "data": 3
      },
      {
        "operator": "D",
        "position": 18,
        "data": 2
      },
      {
        "operator": "D",
        "position": 25,
        "data": 3
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 4
  },
  {
    "tags": {},
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 30,
    "alignmentStart": 1,
    "readGroupId": -1,
    "readName": "xD*",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "X",
        "position": 1,
        "data": 0
      },
      {
        "operator": "X",
        "position": 9,
        "data": 3
      },
      {
        "operator": "D",
        "position": 10,
        "data": 1
      },
      {
        "operator": "X",
        "position": 10,
        "data": 3
      },
      {
        "operator": "X",
        "position": 17,
        "data": 2
      },
      {
        "operator": "D",
        "position": 18,
        "data": 2
      },
      {
        "operator": "X",
        "position": 18,
        "data": 3
      },
      {
        "operator": "X",
        "position": 19,
        "data": 2
      },
      {
        "operator": "D",
        "position": 25,
        "data": 3
      },
      {
        "operator": "X",
        "position": 30,
        "data": 0
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 5
  },
  {
    "tags": {},
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 20,
    "alignmentStart": 1,
    "readGroupId": -1,
    "readName": "xN",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "N",
        "position": 11,
        "data": 20
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 6
  },
  {
    "tags": {},
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 20,
    "alignmentStart": 1,
    "readGroupId": -1,
    "readName": "xN*",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "X",
        "position": 1,
        "data": 0
      },
      {
        "operator": "X",
        "position": 9,
        "data": 0
      },
      {
        "operator": "X",
        "position": 10,
        "data": 0
      },
      {
        "operator": "N",
        "position": 11,
        "data": 20
      },
      {
        "operator": "X",
        "position": 11,
        "data": 0
      },
      {
        "operator": "X",
        "position": 12,
        "data": 0
      },
      {
        "operator": "X",
        "position": 20,
        "data": 0
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 7
  },
  {
    "tags": {
      "NM": 2,
      "MD": "9N0N9"
    },
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 40,
    "alignmentStart": 11,
    "readGroupId": -1,
    "readName": "xS",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "H",
        "position": 1,
        "data": 5
      },
      {
        "operator": "S",
        "position": 1,
        "data": "AAAAAAAAAA"
      },
      {
        "operator": "S",
        "position": 31,
        "data": "TTTTTTTTTT"
      },
      {
        "operator": "H",
        "position": 41,
        "data": 5
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 8
  },
  {
    "tags": {
      "NM": 2,
      "MD": "9N0N9"
    },
    "flags": 0,
    "cramFlags": 3,
    "sequenceId": 0,
    "readLength": 40,
    "alignmentStart": 11,
    "readGroupId": -1,
    "readName": "xS*",
    "templateSize": 0,
    "readFeatures": [
      {
        "operator": "H",
        "position": 1,
        "data": 5
      },
      {
        "operator": "S",
        "position": 1,
        "data": "CAAAATAAAA"
      },
      {
        "operator": "S",
        "position": 31,
        "data": "TTTTATTTTC"
      },
      {
        "operator": "H",
        "position": 41,
        "data": 5
      }
    ],
    "mappingQuality": 255,
    "qualityScores": [
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255,
      255
    ],
    "uniqueId": 9
  }
]